examples/nlp/text_generation_fnet.py

```diff
@@ -0,0 +1,393 @@
+    sentence1 = tokenizer.texts_to_sequences(inputs)
+    sentence2 = tokenizer.texts_to_sequences(outputs)
+
+    for (s1, s2) in zip(sentence1, sentence2):
+        # Taking only those sentences that are less than the maximum allowed length
+        if len(s1) <= MAX_LENGTH and len(s2) <= MAX_LENGTH:
+            # Adding the start and end tokens
+            tokenized_inputs.append(START_TOKEN + s1 + END_TOKEN)
+            tokenized_outputs.append(START_TOKEN + s2 + END_TOKEN)
+
+    # Padding sequences
+    tokenized_inputs = sequence.pad_sequences(
+        tokenized_inputs, maxlen=MAX_LENGTH, padding="post"
+    )
+    tokenized_outputs = sequence.pad_sequences(
+        tokenized_outputs, maxlen=MAX_LENGTH + 1, padding="post"
+    )
+
+    return tokenized_inputs, tokenized_outputs
+
+
+questions, answers = tokenize_and_filter(questions, answers)
+
+
+# Creating a tf.data.Dataset
+dataset = tf.data.Dataset.from_tensor_slices(
+    (
+        {"encoder_inputs": questions, "decoder_inputs": answers[:, :-1]},
+        {"outputs": answers[:, 1:]},
+    )
+)
+
+dataset = dataset.cache().shuffle(BUFFER_SIZE)
+dataset = dataset.batch(BATCH_SIZE).prefetch(tf.data.experimental.AUTOTUNE)
+
+"""
+## Creating the FNet Encoder
+
+The FNet paper proposes a replacement for the standard attention mechanism used by the
+Transformer architecture (Vaswani et al., 2017).
+
+![Architecture](https://i.imgur.com/rLg47qU.png)
+
+The outputs of the FFT layer are complex in nature. To avoid processing complex layers,
+only the real part(magnitude) is extracted.
+
+The dense layers that follow the Fourier transformation act as convolutions applied on
+the frequency domain.
+"""
+
+
+class TransformerEncoder(layers.Layer):
+    def __init__(self, embed_dim, dense_dim, **kwargs):
+        super(TransformerEncoder, self).__init__(**kwargs)
+        self.embed_dim = embed_dim
+        self.dense_dim = dense_dim
+        self.dense_proj = keras.Sequential(
+            [layers.Dense(dense_dim, activation="relu"), layers.Dense(embed_dim),]
+        )
+        self.layernorm_1 = layers.LayerNormalization()
+        self.layernorm_2 = layers.LayerNormalization()
+
+    def call(self, inputs):
+        # Casting the inputs to complex64
+        inp_complex = tf.cast(inputs, tf.complex64)
+        # Projecting the inputs to the frequency domain using FFT2D and
+        # extracting the real part of the output
+        fft = tf.math.real(tf.signal.fft2d(inp_complex))
+        proj_input = self.layernorm_1(inputs + fft)
+        proj_output = self.dense_proj(proj_input)
+        return self.layernorm_2(proj_input + proj_output)
+
+
+"""
+## Creating the Decoder
+The decoder architecture remains the same as the one proposed by (Vaswani et al., 2017)
+in the original transformer architecture, consisting of an embedding, positional
+encoding, two masked multiheaded attention layers and finally the dense output layers
+
+
+"""
+
+
+class PositionalEmbedding(layers.Layer):
```

Comment:
If PositionalEmbedding and TransformerDecoder are taken from the Deep Learning with Python implementation (https://keras.io/examples/nlp/neural_machine_translation_with_transformer/), you should mention that.

Reply:
I had cited the machine translation example at the bottom of the notebook but since you have mentioned it here, I have added it here as well.